tests/middleware/test_message_state.py

```diff
@@ -0,0 +1,79 @@
+import remoulade
+from remoulade.state.middleware import MessageState
+from remoulade.cancel import Cancel
+from remoulade.middleware import SkipMessage
+from remoulade.middleware import Middleware
+
+
+class TestMessageState:
+    """Class to test the middleware
+    MessageState
+    """
+
+    @staticmethod
+    @remoulade.actor
+    def add(x, y):
+        return x + y
+
+    @staticmethod
+    @remoulade.actor
+    def error(x, y):
+        raise Exception()
+
+    """ TODO: find alternative to change priority of rabbitmq_broker in order to
+           use it in this test"""
+    def test_pending_state_message(self, rabbitmq_state, redis_state_backend):
+        rabbitmq_state.declare_actor(TestMessageState.add)
+        rabbitmq_state.add_middleware(MessageState(backend=redis_state_backend))
+        msg = TestMessageState.add.send(5, 7)
+        assert redis_state_backend._get(msg.message_id)['state'] == "Pending"
+
+    def test_success_state_message(self, local_broker, redis_state_backend):
+        local_broker.declare_actor(TestMessageState.add)
+        local_broker.add_middleware(MessageState(backend=redis_state_backend))
+        msg = TestMessageState.add.send(5, 7).result
+        assert redis_state_backend._get(msg.message_id)['state'] == "Success"
+
+    def test_started_state_message(self, local_broker, redis_state_backend):
+        local_broker.declare_actor(TestMessageState.add)
+        local_broker.add_middleware(MessageState(backend=redis_state_backend))
+        msg = TestMessageState.add.send(5, 7)
+        local_broker.emit_before("process_message", msg)
+        assert redis_state_backend._get(msg.message_id)['state'] == "Started"
+
+    def test_failure_state_message(self, stub_broker, stub_worker, redis_state_backend):
+        stub_broker.declare_actor(TestMessageState.add)
+        stub_broker.add_middleware(MessageState(backend=redis_state_backend))
+        remoulade.declare_actors([TestMessageState.error])
+        msg = TestMessageState.error.send()
+        stub_broker.join(TestMessageState.error.queue_name)
+        stub_worker.join()
+        assert redis_state_backend._get(msg.message_id)['state'] == "Failure"
+
+    def test_cancel_state_message(self, stub_broker, stub_worker,
+                                  redis_state_backend, redis_cancel_backend):
+        stub_broker.add_middleware(Cancel(backend=redis_cancel_backend))
+        stub_broker.add_middleware(MessageState(backend=redis_state_backend))
+        remoulade.declare_actors([TestMessageState.add])
+        stub_worker.pause()
```

Comment:
If you take a look at other tests in the app, you can see that there is some comment explaining what's happening. It's not mandatory but here you can add a comment explaining why you pause the worker (as it's not simple)